File: LegalDocParser/LegalDocParser/Models/ConversionOptions.cs
using LegalDocParser.Models.Enums;

namespace LegalDocParser.Models;

/// <summary>
/// PDF 변환 옵션
/// </summary>
public record ConversionOptions
{
    public OutputFormat Format { get; init; } = OutputFormat.Markdown;
    public TableExtractionMode TableMode { get; init; } = TableExtractionMode.Local;
    public OcrLanguage OcrLang { get; init; } = OcrLanguage.None;
    public bool UseTaggedPdf { get; init; } = false;
    public bool EnableAiSafetyFilter { get; init; } = true;
    public string? PageRange { get; init; } = null;
}

File: LegalDocParser/models/conversion_options.py
from dataclasses import dataclass, field

from .enums import OutputFormat, TableExtractionMode, OcrLanguage


@dataclass
class ConversionOptions:
    format: OutputFormat = OutputFormat.MARKDOWN
    table_mode: TableExtractionMode = TableExtractionMode.LOCAL
    ocr_lang: OcrLanguage = OcrLanguage.NONE
    use_tagged_pdf: bool = False
    enable_ai_safety_filter: bool = True
    page_range: str = ""
